packages/inquirer/inquirer.test.mts

```diff
@@ -760,6 +771,39 @@ describe('inquirer.prompt(...)', () => {
   });
 });
 
+describe('AbortSignal support', () => {
+  it('throws on aborted signal', async () => {
+    const localPrompt = inquirer.createPromptModule({ signal: AbortSignal.abort() });
+    localPrompt.registerPrompt('stub', StubEventualyFailingPrompt);
+
+    const promise = localPrompt({ type: 'stub', name: 'q1', message: 'message' });
+    await expect(promise).rejects.toThrow(AbortPromptError);
+  });
+
+  it('legacy prompts can be aborted', async () => {
+    const abortController = new AbortController();
+    const localPrompt = inquirer.createPromptModule({ signal: abortController.signal });
+    localPrompt.registerPrompt('stub', StubEventualyFailingPrompt);
+
+    const promise = localPrompt({ type: 'stub', name: 'q1', message: 'message' });
+    abortController.abort();
+    await expect(promise).rejects.toThrow(AbortPromptError);
+  });
+
+  it('modern prompts can be aborted', async () => {
+    const abortController = new AbortController();
+    const localPrompt = inquirer.createPromptModule({ signal: abortController.signal });
```

Comment:
Interface wise, wouldn't it make more sense to pass signal to `localPrompt([...], { signal })`?

It feels a bit weird to me to abort the module and not a specific prompt session.

Reply:
That's the Modern implementation which makes sense in modern api.
In yeoman-environment, when a failure happens we need to cancel any prompt that is been showing.

Reply:
Since `yeoman-environment` wrap inquirer with the `prompt` method, couldn't it inject the signal there instead?

Reply:
Inquirer is wrapped by `@yoeman/adapter` and the only reason is to close the prompt:
https://github.com/yeoman/yeoman-api/blob/7c9c24374b1b166bd01b3dbcfdd6ce6976de87b3/workspaces/adapter/src/adapter.ts#L86-L92

Wrapping can be dropped if the signal is added in the module instantiation:
https://github.com/yeoman/yeoman-api/blob/7c9c24374b1b166bd01b3dbcfdd6ce6976de87b3/workspaces/adapter/src/adapter.ts#L44-L48

That's the issue motivation.

An additional signal can be added to the api using `AbortSignal.any()` to merge both signals.

Reply:
I don't understand what the issue would be to change where the signal is passed to:

```js
const promptPromise = this.promptModule(questions, initialAnswers, { signal });
```

Can you take some time explain why this won't work in Yeoman? (or the amount of extra complexity this adds if that's the issue?)

Reply:
`@yeoman/adapter` lifecycle is 1 instance per execution.
A single adapter can be shared between many environments. So a prompt will block inputs and queue logs for every environment.

An adapter creates a single inquirer module instance. So:
- the inquirer should be canceled on adapter close at any point. And new prompt will be denied to be shown unless a new inquirer module is created.
- a per execution cancel requires the prompt call to be wrapped, like it is now. There is no reason for this breaking change if the wrapping cannot be dropped.

Reply:
I reflected a bit on this. For `inquirer` as a library, I definitively think it only make sense from the API perspective to abort a "prompt session" and rather than the whole module.

```js
inquirer.prompt([...], { signal });

// or
const prompt = inquirer.createPromptModule();
prompt([...], { signal });
```

Even if this leads to a breaking change in Yeoman, it's better to surface a coherent API in `inquirer`. But I'm pretty sure you could plaster over that and maintain the API backward compatible (like store the signal on the class, and pass it to every prompt call)

Reply:
@SBoudrias I agree about the default provided PromptModule.
Not so much about `createPromptModule` because that's a completely different use case.
As you can see in https://github.com/SBoudrias/Inquirer.js/pull/1535#issuecomment-2325379295.
AbortSignal support has been added to `createPromptModule` with Modern prompts just by adding support to the core.

Reply:
Ah see, this is what I hadn't realized. `inquirer` is prettttty old code, so I'm not so familiar anymore with every details. I didn't remember the options where passed to `createPromptModule` already! - And so truly didn't understand the conflict with this.

I'm torn between:
1. Keeping the module level abort then, since it works out of the box with the built-in prompts. And then let's bring back the module level abort check, since without it the behaviour won't be consistent with plugins. (so having 2 interfaces: `createPromptModule` and `inquirer.prompt`)
2. Drop the `signal` parameters at the prompt module level (ignore it + disallow in type). And only allow within the `prompt` function.

Any preference? Since it pre-exist, I'm not feeling super strongly anymore about either options.

Reply:
I really want to keep the signal at module level.

Current PR adds module level and method level.

I suggest to drop at method level for simplicity.
And change `inquirer.prompt` to create a new module each time:
```js
inquirer.prompt = (question, answers, options) => {
  return createModulePrompt(options)(questions, answers);
}
```

Reply:
Ah yeah okay, that direction works for me.